File: py800mon/monitor/uierror.py
import time

from ..actions import Actions
from ..app import Component
from .appstate import state


class UIErrorTimeout(Component):
    async def update(self):
        if not state.ui_error:
            return False
        deadline = state.ui_error_deadline
        if deadline and time.monotonic() >= deadline:
            self.app.dispatch_action(Actions.CLEAR_UI_ERROR)
            return True
        return False

File: go800mon/a800mon/monitor/uierror.go
package monitor

import (
	"context"
	"time"
)

type UIErrorTimeout struct {
	BaseComponent
}

func NewUIErrorTimeout() *UIErrorTimeout {
	return &UIErrorTimeout{}
}

func (u *UIErrorTimeout) Update(_ctx context.Context) (bool, error) {
	st := State()
	if st.UIError == "" {
		return false, nil
	}
	if !st.UIErrorUntil.IsZero() && time.Now().After(st.UIErrorUntil) {
		u.App().DispatchAction(ActionClearUIError, nil)
		return true, nil
	}
	return false, nil
}
